external-api/api-tests/v1/eStore.api-test.ts

```diff
@@ -23,198 +36,123 @@ jest.mock('../../src/database/mongo-client', () => ({
-        destinationMarket: 'UK',
-        riskMarket: '1',
+  describe('When the Mongo deal is not valid', () => {
+    it('Should return 500 with catch-all error message', async () => {
+      const invalidPayload = {
+        ...payload,
+        dealId: 'invalid',
       };
 
-      const response = await createFacilityFolder('../../etc', '0071029412', createFacilityFolderPayload);
+      const { status, body } = await post(invalidPayload).to('/estore');
 
-      expect(response.status).toEqual(400);
+      expect(status).toEqual(HttpStatusCode.InternalServerError);
+      expect(body.message).toEqual('Unable to create eStore directories');
     });
+  });
 
-    it('should return an error response if dealIdentifier is invalid', async () => {
-      const createFacilityFolderPayload = {
-        dealIdentifier: 'testDeal',
-        facilityIdentifier: 'testFacility',
-        exporterName: 'testName',
-        buyerName: 'testBuyer',
-        destinationMarket: 'UK',
-        riskMarket: '1',
-      };
-
-      const response = await createFacilityFolder('00329453', '../../etc', createFacilityFolderPayload);
+  describe('eStore CRON jobs', () => {
+    it('Should return 201 for a new deal payload', async () => {
+      const { status } = await post(payload).to('/estore');
 
-      expect(response.status).toEqual(400);
+      expect(status).toEqual(HttpStatusCode.Created);
     });
 
-    it('should return an ok response if siteId and dealIdentifier are valid', async () => {
-      const createFacilityFolderPayload = {
-        dealIdentifier: 'testDeal',
-        facilityIdentifier: 'testFacility',
-        exporterName: 'testName',
-        buyerName: 'testBuyer',
-        destinationMarket: 'UK',
-        riskMarket: '1',
-      };
+    it('Should create a new entry cron-job-logs collection for a new payload', async () => {
+      await post(payload).to('/estore');
 
-      const response = await createFacilityFolder('00329453', '0071029412', createFacilityFolderPayload);
+      // Look up for the deal ID in the collection
+      expect(mockFindOne).toHaveBeenCalled();
 
-      expect(response.status).toEqual(200);
+      // Insert a new entry in the collection
+      expect(mockInsertOne).toHaveBeenCalled();
     });
-  });
 
-  describe(`api.uploadSupportingDocuments`, () => {
-    it('should return an error response if siteId is invalid', async () => {
-      const uploadSupportingDocumentsPayload = {
-        buyerName: 'testBuyer',
-        documentType: 'testType',
-        fileName: 'testFile',
-        fileLocationPath: 'testLocation',
-      };
-      const response = await uploadSupportingDocuments('../../etc', '0071029412', uploadSupportingDocumentsPayload);
-
-      expect(response.status).toEqual(400);
-    });
+    it('Should not create a new entry cron-job-logs collection for an existing payload', async () => {
+      mockFindOne.mockResolvedValueOnce({ payload });
+      mockInsertOne.mockReset();
```

Comment:
Is this something we should do in a beforeEach or something? why are re resetting at the start of an individual test - is it being changed from a prior test?